src/core/qgsrendercontext.cpp

```diff
@@ -359,3 +384,24 @@ double QgsRenderContext::convertFromMapUnits( double sizeInMapUnits, QgsUnitType
   }
   return 0.0;
 }
+
+double QgsRenderContext::convertMetersToMapUnits( double meters ) const
+{
+  switch ( mDistanceArea.sourceCrs().mapUnits() )
+  {
+    case QgsUnitTypes::DistanceMeters:
+      return meters;
+      break;
+    case QgsUnitTypes::DistanceDegrees:
+    {
+      QgsPoint pointCenter = mExtent.center();
+      pointCenter = mCoordTransform.transform( pointCenter );
+      return mDistanceArea.measureLineProjected( pointCenter, meters );
+    }
+    break;
+    default:
```

Comment:
can you please list all the enum values you specifically want caught here and remove the default?

I eventually want us to remove all default use from switches as it makes code fragile when new enum values are added - instead we should get the build warning as an indicator that the new enum value needs handling here